helix-core/src/shellwords.rs

```diff
@@ -333,4 +334,9 @@ mod test {
         assert_eq!(Shellwords::from(":o a").parts(), &[":o", "a"]);
         assert_eq!(Shellwords::from(":o a\\ ").parts(), &[":o", "a\\"]);
     }
+
+    #[test]
+    fn test_multibyte_at_end() {
+        assert_eq!(Shellwords::from("𒀀").parts(), &["𒀀"]);
```

Comment:
I liked your examples in the PR description, let's add cases for them to make sure we don't have regressions in the future 👍

```suggestion
        assert_eq!(Shellwords::from("𒀀").parts(), &["𒀀"]);
        assert_eq!(Shellwords::from(":sh echo 𒀀").parts(),  &[":sh", "echo", "𒀀"]);
        assert_eq!(Shellwords::from(":sh echo hello𒀀").parts(),  &[":sh", "echo", "hello𒀀"]);
```

Reply:
I added the change as a separate commit as it is generally cleaner to review (as opposed to an amend). I can squash them if you want.  What is the general preference for this project?

Reply:
reviewers generally squash themselves for small PRs like this if they find it necessary.  I don't think it matters much either way,